internal/util-logging/src/test/scala/ManagedLoggerSpec.scala

```diff
@@ -50,6 +57,28 @@ class ManagedLoggerSpec extends FlatSpec with Matchers {
     log.infoEvent(Vector(Vector(1, 2, 3)))
   }
 
+  it should "be thread safe" in {
+    import java.util.concurrent.{ Executors, TimeUnit }
+    val pool = Executors.newFixedThreadPool(100)
+    for {
+      i <- 1 to 10000
+    } {
+      pool.submit(new Runnable {
+        def run(): Unit = {
+          val stringTypeTag = StringTypeTag[List[Int]]
+          val log = LogExchange.logger(s"foo$i")
+          LogExchange.bindLoggerAppenders(s"foo$i", List(LogExchange.asyncStdout -> Level.Info))
+          if (i % 100 == 0) {
+            log.info(s"foo$i test $stringTypeTag")
+          }
+          Thread.sleep(1)
+        }
+      })
+    }
+    pool.shutdown
+    pool.awaitTermination(30, TimeUnit.SECONDS)
```

Comment:
You mentioned "Trying to reproduce" in the commit message. Does this test reproduce the issue?

Reply:
No, I was not able to reproduce the NPE, but I am keeping the test to test thread safety.